jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/AuthenticationState.java

```diff
@@ -270,6 +271,45 @@ public String toString()
         }
     };
 
+    /**
+     * The {@link SecurityHandler} will use this to wrap the {@link Request}.
+     * And then will return a {@link Deferred} authentication to bypass security constraints.
+     */
+    class ServeAs implements AuthenticationState
+    {
+        private final HttpURI _uri;
+
+        public ServeAs(HttpURI uri)
+        {
+            _uri = uri;
+        }
+
+        public Request wrap(Request request)
+        {
+            if (request.getHttpURI().equals(_uri))
+                return request;
+
+            return new Request.Wrapper(request)
+            {
+                @Override
+                public HttpURI getHttpURI()
+                {
+                    return _uri;
+                }
+            };
+        }
+
+        /**
+         * This interface can be used inside a {@link ServeAs} implementation to wrap the request
+         * changing its target to a given path. If a {@link Request} implements this interface it can
+         * be obtained with the {@link Request#as(Request, Class)} method.
+         */
+        public interface Path
```

Comment:
Path is a little overloaded here.  Maybe `Wrapper` would be better (but still overloaded).
```suggestion
        public interface Wrapper
```

Reply:
Changed it to `PathWrapper` for now which doesn't conflict with any other classes.

But I don't have a strong preference on the naming, so can change it if there is a better name.